rct229/ruleset_functions/baseline_systems/is_baseline_system_3.py

```diff
@@ -0,0 +1,113 @@
+    is_hvac_sys_cooling_type_dx,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_cooling_type_fluid_loop import (
+    is_hvac_sys_cooling_type_fluid_loop,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_fan_sys_CV import (
+    is_hvac_sys_fan_sys_cv,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_fluid_loop_purchased_CHW import (
+    is_hvac_sys_fluid_loop_purchased_chw,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_fluid_loop_purchased_heating import (
+    is_hvac_sys_fluid_loop_purchased_heating,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_heating_type_fluid_loop import (
+    is_hvac_sys_heating_type_fluid_loop,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_heating_type_furnace import (
+    is_hvac_sys_heating_type_furnace,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import (
+    HVAC_SYS,
+    find_exactly_one_hvac_system,
+)
+
+HEATING_SYSTEM = schema_enums["HeatingSystemOptions"]
+
+
+def is_baseline_system_3(rmi_b, hvac_b_id, terminal_unit_id_list, zone_id_list):
+    """
+    Get either Sys-3, Sys-3a, Sys-3b, Sys-3c, or Not_Sys_3 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 3 (PSZ), system 3a (system 3 with purchased CHW),
+    system 3b (system 3 with purchased heating), system 3c (system 3 with purchased CHW and purchased HW).
+
+    Parameters
+    ----------
+    rmi_b json
+        To evaluate if the hvac system is modeled as either Sys-3, Sys-3a, Sys-3b, Sys-3c, or Not_Sys_3 in the B_RMI.
+
+    hvac_b_id list
+        The id of the hvac system to evaluate.
+
+    terminal_unit_id_list
+        List list of terminal unit IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+
+    zone_id_list list
+        list of zone IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+
+    Returns
+    -------
+    The function returns either Sys-3, Sys-3a, Sys-3b, Sys-3c, or Not_Sys_3 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 3 (PSZ),
+    system 3a (system 3 with purchased CHW), system 3b (system 3 with purchased heating), system 3c (system 3 with purchased CHW and purchased HW).
+    """
+    is_baseline_system_3 = HVAC_SYS.UNMATCHED
+
+    hvac_b = find_exactly_one_hvac_system(rmi_b, hvac_b_id)
+    # check if the hvac system has the required sub systems for system type 3
+    preheat_system = hvac_b.get("preheat_system")
+    has_required_sys = (
+        preheat_system is None
+        or preheat_system.get("heating_system_type") is None
```

Comment:
Shouldn't these be `and`s instead of `or`s?
If so, the RDS should be updated.

Reply:
No, this is correct as is. I was thinking backwards.

Reply:
Thanks for verifying! I left it as is